test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/SparseFieldSets/SparseFieldSetTests.cs

```diff
@@ -0,0 +1,273 @@
+using FluentAssertions;
+using JsonApiDotNetCore.Configuration;
+using JsonApiDotNetCore.Serialization.Objects;
+using JsonApiDotNetCore.Services;
+using JsonApiDotNetCoreMongoDbExample;
+using JsonApiDotNetCoreMongoDbExample.Models;
+using Microsoft.Extensions.DependencyInjection;
+using MongoDB.Driver;
+using Xunit;
+
+namespace JsonApiDotNetCoreMongoDbExampleTests.IntegrationTests.SparseFieldSets
+{
+    public sealed class SparseFieldSetTests : IClassFixture<IntegrationTestContext<Startup>>
+    {
+        private readonly IntegrationTestContext<Startup> _testContext;
+        private readonly Faker<User> _userFaker;
+
+        public SparseFieldSetTests(IntegrationTestContext<Startup> testContext)
+        {
+            _testContext = testContext;
+
+            testContext.ConfigureServicesAfterStartup(services =>
+            {
+                services.AddSingleton<ResourceCaptureStore>();
+
+                services.AddResourceRepository<ResultCapturingRepository<Blog>>();
+                services.AddResourceRepository<ResultCapturingRepository<Article>>();
+                services.AddResourceRepository<ResultCapturingRepository<Author>>();
+                services.AddResourceRepository<ResultCapturingRepository<TodoItem>>();
+
+                services.AddScoped<IResourceService<Article, string>, JsonApiResourceService<Article, string>>();
+            });
+            
+            _userFaker = new Faker<User>()
+                .RuleFor(u => u.UserName, f => f.Internet.UserName())
+                .RuleFor(u => u.Password, f => f.Internet.Password());
+        }
+
+        [Fact]
+        public async Task Can_select_fields_in_primary_resources()
+        {
+            // Arrange
+            var store = _testContext.Factory.Services.GetRequiredService<ResourceCaptureStore>();
+            store.Clear();
+
+            var article = new Article
+            {
+                Caption = "One",
+                Url = "https://one.domain.com"
+            };
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                var collection = db.GetCollection<Article>(nameof(Article));
+                await collection.DeleteManyAsync(Builders<Article>.Filter.Empty);
+                await collection.InsertOneAsync(article);
+            });
+
+            var route = "/api/v1/articles?fields[articles]=caption"; // TODO: once relationships are implemented select author field too
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<Document>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.OK);
+
+            responseDocument.ManyData.Should().HaveCount(1);
+            responseDocument.ManyData[0].Id.Should().Be(article.StringId);
+            responseDocument.ManyData[0].Attributes.Should().HaveCount(1);
+            responseDocument.ManyData[0].Attributes["caption"].Should().Be(article.Caption);
+
```

Comment:
Undo the removal of relationship assertions in this file. Do assert they are null.